cmd/fund/fund.go

```diff
@@ -0,0 +1,268 @@
+	var wg sync.WaitGroup
+	wg.Add(concurrency)
+
+	// Function to fund wallets
+	fundWallet := func() {
+		defer wg.Done()
+		for wallet := range walletChan {
+			nonce, err := generateNonce(web3Client)
+			if err != nil {
+				log.Error().Err(err).Msg("Error getting nonce")
+				return
+			}
+
+			// Fund the wallet using the obtained nonce
+			_, err = web3Client.Eth.SyncSendRawTransaction(
+				wallet.Address,
+				amountWei,
+				nonce,
+				walletFundingGas,
+				web3Client.Utils.ToGWei(1),
+				nil,
+			)
+			if err != nil {
+				log.Error().Err(err).Str("wallet", wallet.Address.Hex()).Msg("Error funding wallet")
+				return
+			}
+
+			log.Info().Str("wallet", wallet.Address.Hex()).Msgf("Funded with %s wei", amountWei.String())
+		}
+	}
+
+	// Start funding the wallets concurrently
+	for i := 0; i < concurrency; i++ {
+		go fundWallet()
+	}
+
+	// Wait for all goroutines to finish
+	wg.Wait()
+	return nil
+}
+
+// fundCmd represents the fund command
+var FundCmd = &cobra.Command{
+	Use:   "fund",
+	Short: "Bulk fund many crypto wallets automatically.",
+	Long:  usage,
+	Run: func(cmd *cobra.Command, args []string) {
+		if err := runFunding(cmd); err != nil {
+			log.Error().Err(err).Msg("Error funding wallets")
+		}
+	},
+}
+
+func runFunding(cmd *cobra.Command) error {
+	// Capture the start time
+	startTime := time.Now()
+
+	// Remove '0x' prefix from fundingWalletPK if present
+  fundingWalletPK = strings.TrimPrefix(fundingWalletPK, "0x")
+
+	// setup new web3 session with remote rpc node
+	web3Client, clientErr := web3.NewWeb3(chainRPC)
+	if clientErr != nil {
+		cmd.PrintErrf("There was an error creating web3 client: %s", clientErr.Error())
+		return clientErr
+	}
+
+	// add pk to session for sending signed transactions
+	if setAcctErr := web3Client.Eth.SetAccount(fundingWalletPK); setAcctErr != nil {
+		cmd.PrintErrf("There was an error setting account with pk: %s", setAcctErr.Error())
+		return setAcctErr
+	}
+
+	// set proper chainId for corresponding chainRPC
+	cdkChainId := int64(chainID)
+	web3Client.Eth.SetChainId(cdkChainId)
+
+	// generate set of new wallet objects
+	wallets, genWalletErr := generateWallets(walletCount)
+	if genWalletErr != nil {
+		cmd.PrintErrf("There was an error generating wallet objects: %s", genWalletErr.Error())
+		return genWalletErr
+	}
+
+	// fund all crypto wallets
+	log.Info().Msg("Starting to fund loadtest wallets...")
+	fundWalletErr := fundWallets(web3Client, wallets, big.NewInt(int64(walletFundingAmt*1e18)), uint64(walletFundingGas), concurrencyLevel)
+	if fundWalletErr != nil {
+		log.Error().Err(fundWalletErr).Msg("Error funding wallets")
+		return fundWalletErr
+	}
+
+	// Save wallet details to a file
+	outputFile := outputFileFlag // You can modify the file format or name as needed
+
+	type WalletDetails struct {
+		Address    string `json:"Address"`
+		PrivateKey string `json:"PrivateKey"`
+	}
+
+	walletDetails := make([]WalletDetails, len(wallets))
+	for i, w := range wallets {
```

Comment:
Could you also log the wallet public/private keys?

Reply:
tried this, believe it introduces too noisy logs and user can simply cat resulting keys.json if they want to review wallet details is my thinking